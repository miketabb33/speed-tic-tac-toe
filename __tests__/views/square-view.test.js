import React from 'react'
import renderer from 'react-test-renderer'
import SquareView from '../../views/square-view.js'

const whateverIndex = 1

test('render empty', () => {
	const subject = renderer.create(
		<SquareView 
			onClick = { () => {} }
			squareIndex = { whateverIndex }
			boardIndex = { whateverIndex }
		/>
	)

	const result = subject.toJSON()
	expect(result).toMatchSnapshot()
})

test('render x', () => {
	const subject = renderer.create(
		<SquareView 
		 	value = "x"
			onClick = { () => {} }
			squareIndex = { whateverIndex }
			boardIndex = { whateverIndex }
		/>
	)

	const result = subject.toJSON()
	expect(result).toMatchSnapshot()
})

test('render o', () => {
	const subject = renderer.create(
		<SquareView 
		 	value = "o"
			onClick = { () => {} }
			squareIndex = { whateverIndex }
			boardIndex = { whateverIndex }
		/>
	)

	const result = subject.toJSON()
	expect(result).toMatchSnapshot()
})