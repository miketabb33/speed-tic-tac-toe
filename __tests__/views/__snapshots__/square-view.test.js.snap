// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render empty 1`] = `
<button
  className="square"
  id="board-2-square-2"
  onClick={[Function]}
>
  <svg
    height={79}
    width={79}
  >
    <rect
      className="clear"
      height={79}
      width={79}
    />
  </svg>
</button>
`;

exports[`render o 1`] = `
<button
  className="square"
  id="board-2-square-2"
  onClick={[Function]}
>
  <svg
    height={79}
    width={79}
  >
    <rect
      className="clear"
      height={79}
      width={79}
    />
  </svg>
</button>
`;

exports[`render x 1`] = `
<button
  className="square"
  id="board-2-square-2"
  onClick={[Function]}
>
  <svg
    height={79}
    width={79}
  >
    <rect
      className="clear"
      height={79}
      width={79}
    />
  </svg>
</button>
`;
